ccache.c

```diff
@@ -2441,6 +2441,31 @@ cc_process_args(struct args *args, struct args **preprocessor_args,
 			free(option);
 			continue;
 		}
+                if (str_startswith(argv[i], "-B")) {
+			char *relpath = make_relative_path(x_strdup(argv[i] + 2));
+			char *option = format("-B%s", relpath);
+			args_add(stripped_args, option);
+			free(relpath);
+			free(option);
+			continue;
+                }
+                if (str_startswith(argv[i], "-iprefix")) {
+			char *relpath = make_relative_path(x_strdup(argv[i+1]));
```

Comment:
I RTFM'ed GCC and clearly there is a space between iprefix and the directory as mentioned here 
https://gcc.gnu.org/onlinedocs/cpp/Invocation.html
> 
-iprefix prefix
    Specify prefix as the prefix for subsequent -iwithprefix options. If the prefix represents a directory, you should include the final ‘/’. 
>
Also,  the link you've provided is broken. Can you please paste the compiler invocation here ?